src/btree/col_modify.c

```diff
@@ -79,11 +79,13 @@ __wt_col_modify(WT_CURSOR_BTREE *cbt, uint64_t recno, const WT_ITEM *value, WT_U
         }
     } else {
         /* Since on this path we never set append, make sure we aren't appending. */
-        WT_ASSERT(session, recno != WT_RECNO_OOB);
-        WT_ASSERT(session,
+        WT_ASSERT_ALWAYS(
+          session, recno != WT_RECNO_OOB, "Attempting to append when recno is in bounds");
```

Comment:
This one I admit you cannot phrase too well without having more context, perhaps: "An invalid recno was passed in the non-append path."

Reply:
I had a re-read of this one and I think we can use "Out-of-band recno provided for a non-append operation". 

Looking at `recno` and `upd_arg` there's an implied coupling between the two where the valid states are:
1. `recno` is off-page and `upg_arg == NULL` => We're appending a new record to the page
2. `recno` is on-page and `upd_arg != NULL` => We're modifying/inserting something on the page. i.e. anything other than an append

We check these same conditions in the other branch of the `if` and expect them to be false in order to set `append=true` so I think this comes down to input validation for `__wt_col_modify`.

Reply:
"Out-of-bound" but yeah sounds good to me.